### DISCUSSION TITLE
Search docs from another site

### DISCUSSION THREAD

**Discussion by SteveRosam** [squidfunk/mkdocs-material#6490] [General]
I have been asked to investigate how we can search our docs site from our product site.

Is the docs search something that is pre-built and can be extracted from the site after the build?
Is there another way to achieve this?

**Comment by squidfunk** [comment]
Integrating other sites built with Material for MkDocs into a single federated search is tracked in https://github.com/squidfunk/mkdocs-material/issues/5230. If your other site is not built with MkDocs, it's definitely more work, but I'm currently working on the search, so any feedback and use cases are appreciate. Feel free to share a little more in what you would like to solve and how.

__Related__

- https://github.com/squidfunk/mkdocs-material/issues/6307

**Reply by SteveRosam** [reply]
Wow I just read [this](https://github.com/squidfunk/mkdocs-material/issues/6307). That's going to be awesome.
Our particular use case is embedding docs into our product (built with Angular). So we can have docs search right there where the user is having whatever issue they're having. From the mockups I've seen, the design team want search, search results and the content from whatever page is clicked to be displayed in the product (with a link to the docs site in case the user wants to go there)

I guess I would need some kind of API or something to do the search and get the results. Might be better to scrape the built/deployed site and do some vectorization and search that.

**Reply by squidfunk** [reply]
The documentation search will be an entire project of its own, so you will be able to integrate it into your Angluar site. The aim is to solve documentation search not only for Material for MkDocs, but have a portable solution that works in all kinds of technologies.

Note that this is a gargantuan task, and while I already have a pretty mighty prototype, it's still some way to go, also given how much time I have to pour into support, bug fixing and implementing change requests. We really hope for more organizational sponsors to chime in, support my and our (the growing team's) work, so I can focus on search and other features that will move this project onto a whole new level.

**Reply by SteveRosam** [reply]
Thanks for taking time to reply. Sounds like this could be the solution we need.
